File: php/payments.php
<?php

require_once 'entity.php';
require_once 'invoices.php';
require_once 'users.php';
require_once './../php_aux/business_default.php';

class Payment extends Entity
{
    public static $resource = '/payments/';
    public static $json_name = 'payment';

    public function __construct()
    {
        $this->escape_fields = ['payment_type', 'pay_date'];
        $this->json_property('id','integer');
        $this->json_property('note','string');
        $this->json_property('payment_type','integer');
        $this->json_property('pay_date','DateTime');
        $this->json_property('amount','float');
        $this->json_property('invoice', 'Invoice', $many = False,
                             $recursive = True);
        $this->json_property('payment_recorder', 'Users', $many = False,
                             $recursive = True);
        $this->json_property('send_sms','boolean');
        $this->json_property('send_email','boolean');
    }

    public function payment_type_string(){
        /*Return the string value of payment type from the
          common.business_default file
        */
        return PAYMENT_TYPES[$this->payment_type];
    }
}

class Payments extends Resource
{
    public function __construct()
    {
        $this->entity_class = 'Payment';
        $this->json_name = 'payments';
    }
}

$payments = new Payments();

File: python/payments.py
import datetime
import sdk.python.entities
import sdk.python.users
from sdk.python.automatic_payment_relationships import \
    AutomaticPaymentRelationship
from sdk.python.util.business_default import PAYMENT_TYPES
from sdk.python.entities import Property


class Payment(sdk.python.entities.Entity):

    resource = '/payments/'
    json_name = 'payment'

    def __init__(self):
        super(Payment, self).__init__()
        self.escape_fields = ['payment_type', 'pay_date']

    id = Property(int)
    note = Property(str)
    payment_type = Property(int)
    pay_date = Property(datetime.datetime)
    amount = Property(float)
    payment_recorder = Property(sdk.python.users.Users)
    send_sms = Property(bool)
    send_email = Property(bool)
    charged_by_payment_relationship = Property(AutomaticPaymentRelationship)

    def payment_type_string(self):
        """ Return the string value of payment type from the
            sdk.python.util.business_default file
        """
        return PAYMENT_TYPES[self.payment_type]


class Payments(sdk.python.entities.Resource):

    entity_class = Payment
    json_name = 'payments'


payments = Payments()
